chainer_compiler/elichika/parser/vevaluator.py

```diff
@@ -81,18 +72,44 @@ def c(self, value) -> 'AstContext':
 
 class VEvalOption:
     def __init__(self):
-        self.eval_as_written_target = False
+        self.history = []
+        self.flags = {
+            "eval_as_written_target": False,
+            "ignore_branch": False,
+            "for_unroll": False
+        }
+
+        deque(map(lambda flag: setattr(self.__class__, '_' + flag, property(fset=lambda obj, value: VEvalOption.generic_setter(obj, flag, value),
```

Comment:
Why deque? I guess this is to evaluate the lazy sequence and `list` would be enough.

However, I'd recommend you to write this with a simple for-loop. Short loop comprehensions or map+lambda is sometimes improve readability (and it's actually fun to write :). IMHO however, it makes code harder to read or debug. For example, it'll be harder to insert print functions for debugging.

I like the guide of google: https://google.github.io/styleguide/pyguide.html#27-comprehensions--generator-expressions

Reply:
Yes, `list` would be enough I suppose. I was initially using a for loop only, but it had some bug that I couldn't pin point. These two code snippets had different behaviors. The first one worked as expected, but the second one wasn't setting the correct flag on calling `option.<flag>()` (It always was setting `option._for_unroll`  to `True`, probably because it is the last `flag` in the for loop). 

```python
list(map(lambda flag: setattr(self.__class__, '_' + flag, property(fset=lambda obj, value: VEvalOption.generic_setter(obj, flag, value),
                                                                    fget=lambda obj: VEvalOption.generic_getter(obj, flag))),
            self.flags.keys()))
list(map(lambda flag: setattr(self.__class__, flag, auto_set_unset(lambda obj: VEvalOption.generic_setter(obj, flag, True), flag)),
            self.flags.keys()))
```

```python
for flag in self.flags.keys():
    setattr(self.__class__, '_' + flag, property(fset=lambda obj, value: VEvalOption.generic_setter(obj, flag, value),
                                                    fget=lambda obj: VEvalOption.generic_getter(obj, flag)))

    setattr(self.__class__, flag, auto_set_unset(lambda obj: VEvalOption.generic_setter(obj, flag, True), flag))
```

Reply:
I suspected that this behavior was due to the fact that lambda expression behaves differently than what is expected.
```python
>>> x = []
>>> for i in range(10):
...     x.append(lambda: i)
... 
>>> x[0]()
9
```

Reply:
Okay, I will use `list`. `functools.partial` seems like a nice solution, but it changes the argument list of the lambda function, which would be unwanted in this case as the lambda function is being used as setter/getter functions which require a fixed argument list (obj, key, [value]).